### PR TITLE
add flux-security man page references

### PR DESCRIPTION
**Pull request by garlick**
Add references to the new flux-security man pages from the top level landing doc, as well as the flux admin guide.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ The framework consists of a suite of projects, tools, and libraries which may be
    :hidden:
 
    flux-core <https://flux-framework.readthedocs.io/projects/flux-core/en/latest/index.html>
+   flux-secuirty <https://flux-framework.readthedocs.io/projects/flux-security/en/latest/index.html>
```
Comment (grondo) [inline index.rst:38]: typo: `flux-secuirty`

**Review by grondo — APPROVED** [review]
Thanks! Just spotted one typo.

File: index.rst (hunk 1 above)
Comment (garlick) [reply index.rst]: Just rebased on #129 so this commit was dropped.

**Comment by grondo** [comment]
weird, `make linkcheck` is suddenly failing on some github urls:
```

(line   17) broken    https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models - 404 Client Error: Site Not Found for url: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models

(line   30) broken    https://help.github.com/en/github/getting-started-with-github/fork-a-repo - 404 Client Error: Site Not Found for url: https://help.github.com/en/github/getting-started-with-github/fork-a-repo
```

**Comment by garlick** [comment]
Seems to have started working again while I was poking at it so I'll restart CI and see how we do.

**Comment by garlick** [comment]
Thanks for the review - setting MWP.

**Pull request merged**